### PR TITLE
Allow batch explanation on one mask for `CaptumExplainer`

### PR DESCRIPTION
**Pull request by RBendias**
Point 7 in #6375.

### CODE REVIEW

File: test/explain/algorithm/test_captum_explainer.py [hunk 1]
```diff
@@ -103,12 +115,41 @@ def test_captum_explainer_multiclass_classification(
...
+
+@withPackage('captum')
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+def test_captum_explainer_multiclass_classification_batch(
+    node_mask_type,
+    edge_mask_type,
+):
+    index = [2, 3]
+    task_level = 'node'
```
Comment (rusty1s) [inline test/explain/algorithm/test_captum_explainer.py:129]: Pass directly.

File: test/explain/algorithm/test_captum_explainer.py [hunk 2]
```diff
@@ -103,12 +115,41 @@ def test_captum_explainer_multiclass_classification(
...
-        assert 'edge_mask' not in explanation
+
+@withPackage('captum')
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+def test_captum_explainer_multiclass_classification_batch(
+    node_mask_type,
+    edge_mask_type,
+):
+    index = [2, 3]
```
Comment (rusty1s) [inline test/explain/algorithm/test_captum_explainer.py:128]: Doesn't this need to be a `torch.Tensor`?

File: torch_geometric/explain/config.py [hunk 3]
```diff
@@ -18,6 +18,13 @@ class MaskType(Enum):
     attributes = 'attributes'
 
 
+class MaskLevelType(Enum):
```
Comment (rusty1s) [inline torch_geometric/explain/config.py:21]: This feels kinda specific to `captum` - shall we move it there?

File: torch_geometric/nn/models/captum.py [hunk 4]
```diff
@@ -13,8 +13,10 @@
 
 
 def to_captum_model(
-    model: torch.nn.Module, mask_type: str = "edge",
-    output_idx: Optional[int] = None, metadata: Optional[Metadata] = None
+    model: torch.nn.Module,
+    mask_type: str = "edge",
```
Comment (rusty1s) [inline torch_geometric/nn/models/captum.py:17]: `MaskLevelType`?

File: torch_geometric/explain/algorithm/captum_explainer.py [hunk 5]
```diff
@@ -59,6 +60,10 @@ def _get_mask_type(self):
             mask_type = 'edge'
         return mask_type
 
+    def _support_multiple_indices(self):
+        "Check if the method supports multiple indices."
```
Comment (rusty1s) [inline torch_geometric/explain/algorithm/captum_explainer.py:64]:
```suggestion
        r"""Checks if the method supports multiple indices."""
```

File: torch_geometric/explain/algorithm/captum_explainer.py [hunk 6]
```diff
@@ -70,6 +75,20 @@ def forward(
         **kwargs,
     ) -> Explanation:
 
+        # Check if the method supports multiple indices.
+        if isinstance(index, list) and len(index) > 1:
```
Comment (rusty1s) [inline torch_geometric/explain/algorithm/captum_explainer.py:79]: Following the `Explainer` inteface, `index` needs to be `int` or `Tensor`, so we should check via `index.numel() > 1`

File: torch_geometric/explain/algorithm/captum_explainer.py [hunk 7]
```diff
@@ -70,6 +75,20 @@ def forward(
...
+            if not self._support_multiple_indices():
+                raise ValueError(
+                    f"{self.attribution_method.__name__} does not "
+                    "support multiple indices. Please use a single "
+                    "index or a different method.")
+            if self.kwargs.get('internal_batch_size') != 1:
+                warnings.warn("Overriding 'internal_batch_size' to 1")
```
Comment (rusty1s) [inline torch_geometric/explain/algorithm/captum_explainer.py:86]: I don't think we want to warn in case `internal_batch_size` was not set.

File: torch_geometric/explain/algorithm/captum_explainer.py [hunk 8]
```diff
@@ -70,6 +75,20 @@ def forward(
...
+                self.kwargs['internal_batch_size'] = 1
+            warnings.warn(
+                "For all indices, the explanations are computed on the"
+                "same mask. This can lead to overlapping explanations.")
+
```
Comment (rusty1s) [inline torch_geometric/explain/algorithm/captum_explainer.py:91]: That would be the case for every explainer, so I am not sure I understand the warning here.

**Review by rusty1s — APPROVED** [review]
Looks mostly good to me. Thanks!

File: torch_geometric/explain/algorithm/captum_explainer.py [hunk 9]
```diff
@@ -70,6 +75,16 @@ def forward(
         **kwargs,
     ) -> Explanation:
 
+        if isinstance(index, torch.Tensor) and (index.numel() > 1):
```
Comment (rusty1s) [inline torch_geometric/explain/algorithm/captum_explainer.py:78]:
```suggestion
        if isinstance(index, torch.Tensor) and index.numel() > 1:
```

File: torch_geometric/explain/algorithm/captum_explainer.py [hunk 10]
```diff
@@ -70,6 +75,16 @@ def forward(
...
+            if not self._support_multiple_indices():
+                raise ValueError(
+                    f"{self.attribution_method.__name__} does not "
+                    "support multiple indices. Please use a single "
+                    "index or a different method.")
+            if self.kwargs.get('internal_batch_size', 1) != 1:
```
Comment (rusty1s) [inline torch_geometric/explain/algorithm/captum_explainer.py:84]: We probably need to check if that argument is available for the given explainer. Shall we leave a TODO?

**Review by rusty1s — APPROVED** [review]
LGTM.

File: torch_geometric/explain/algorithm/captum_explainer.py (hunk 10 above)
Comment (RBendias) [reply torch_geometric/explain/algorithm/captum_explainer.py:84]: Currently, we only support IG. When adding the support for more methods, I'll adjust those lines (added a reminder in the overall CaptumExplainer issue).
